src/Execute/DeviceViewProvider.ts

```diff
@@ -191,42 +197,59 @@ export class DeviceViewProvider
   }
 
   loadDeviceManager(deviceMan: string, callback: Function): void {
-    const listCmds: Promise<Device[]>[] = [];
     const deviceList: Device[] = [];
+    // load executor environment
+    // 1. the actual target connected to the bridge
+    // 2. simulator on host machine
+    // 3. TODO simulator on docker image
     for (const deviceSpec of supportedSpecs) {
       if (deviceSpec.bridge) {
         const deviceGetCmd = deviceSpec.bridge.deviceListCmd;
+        const listCmds: Promise<Device[]>[] = [];
         listCmds.push(this.getDevicesPromise(deviceGetCmd, deviceSpec));
+        Promise.all(listCmds).then((results) => {
+          for (const result of results) {
+            deviceList.push(...result);
+          }
+        });
       } else {
-        deviceList.push(new Device("HostPC", deviceSpec));
-      }
-    }
-    Promise.all(listCmds).then((results) => {
-      for (const result of results) {
-        deviceList.push(...result);
-      }
-      let executorList: Executor[] = [];
-      const entries = Object.entries(globalBackendMap);
-      for (const entry of entries) {
-        const compiler = entry[1].compiler();
-        if (compiler) {
-          for (const toolchainType of compiler.getToolchainTypes()) {
-            if (compiler.getInstalledToolchains(toolchainType).length > 0) {
-              executorList.push(...entry[1].executors());
-              break;
+        Object.entries(globalBackendMap).map((entry) => {
```

Comment:
```suggestion
        Object.entries(globalBackendMap).map((_backendName, backend) => {
```

How about using `type destructuring` for readability?
I added underbar to _backendName because the value is not used in the loop